src/ChosenSubjects.js

```diff
@@ -1,24 +1,113 @@
+    {
+        subjectsFromJSON[i].id = i;
+    }
+    return subjectsFromJSON;
+}
+
+const getSubjectColumns = subjects => {
+    // priprav sloupce dynamicky z dat
+    var columns = []
+    for (var prop in subjects[0])
+    {
+        if (prop != 'id')
+            columns.push(prop);
+    }
+    // nejak to serad
+    columns.sort();
+    // takhle to DataGrid vyzaduje - sloupce tabulky jsou samostatne objekty
+    columns = columns.map(x => {return {field: x.toString()}});
+    return columns;
+}
+
+
+const updateTotalRow = subjects => {
+    // prepocitej soucty v poslednim radku
+    let totalCredits = 0;
+    let totalLength = 0;
+    for (let i = 0; i < subjects.length; i++)
+    {
+        totalCredits += subjects[i].credits;
+        totalLength += Number(subjects[i]["length"]);
+    }
+
+    return [{id:0, credits:totalCredits, length:totalLength}]
+}
+
 /** Assigned to Hynek */
 // TODO Vypadá to jako job pro https://material-ui.com/components/tables/
 // TODO Tady je example jak dostat data: {subjects[0]["lecturer"]}
 const ChosenSubjects = () => {
-    const subjects = useSubjects();
+    // viz https://reactjs.org/docs/hooks-state.html
+    const [selection, setSelection] = React.useState([]);
+    const [subjectRows, setSubjectRows] = React.useState(getSubjectRows(useSubjects()));
+    const [totalRow, setTotalRow] = React.useState(updateTotalRow(subjectRows));
+
     const classes = useStyles();
     return (
-        <div className={classes.root}>
-            Zde bude vyvíjen seznam vybraných předmětů, kam se umístí až bude potřeba ještě rozmyslíme.
-            Rozhodně to nebude takto přes one-way button "Vybrané předměty".
-            První předmět vyučuje: {subjects[0]["lecturer"]}
+        <div>
+        <div style={{ height: 550, width: '100%' }}>
+            <DataGrid
+                sortModel={sortModel}
+                columns={subjectColumns}
+                rows= {subjectRows}
+                checkboxSelection={true}
+                // zmen vyber
+                onSelectionChange={(newSelection) => {
+                    setSelection(newSelection.rowIds);
+                }}
+                hideFooter={true}
+            />
+        </div>
+        <div>vybrané řádky: {selection.map(rowId => ','+rowId)}</div>
+        <Button
+            variant="outlined"
+            color="secondary"
+            startIcon={<DeleteIcon />}
+            // vymaz vybrane radky (zustavaji ty, ktere nebyly vybrany)
+            onClick={() => {
+                setSubjectRows(
+                    subjectRows.filter(row => !selection.includes(row.id.toString())));
+                setTotalRow(updateTotalRow(subjectRows));
+                setSelection([]);
+            }
+            }>
+            Odstranit vybrané
+        </Button>
+        <div style={{ height: 110, width: '100%' }}>
+            <DataGrid
+                columns={subjectColumns}
+                rows= {totalRow}
+                //rowCount={1}
```

Comment:
Nenechávej v kódu ani zakomentované bloky. Znám to, vím jak vznikají atd. ale snažím se tomu vyhnout. Příště vám můžu ukázat jak pustit debugger :+1: .